File: Behavioral/Observer/Java/SMSSubscriber.java
// SMSSubscriber.java

public class SMSSubscriber extends Subscriber {
  public SMSSubscriber(String name) {
    super(name);
  }

  @Override
  public void update(String news) {
    System.out.println(getName() + " (SMS) received: " + news);
  }
}

File: Behavioral/Observer/Cpp/SMSSubscriber.cpp
// SMSSubscriber.cpp

#include "SMSSubscriber.h"

SMSSubscriber::SMSSubscriber(const std::string& name) : Subscriber(name) {}

void SMSSubscriber::update(const std::string& news) {
  std::cout << name << " (SMS) received: " << news << std::endl;
}
